app/src/modules/battery/battery.c

```diff
@@ -198,7 +199,24 @@ static void battery_task(void)
 			return;
 		}
 
-		if (&TRIGGER_CHAN == chan) {
+		if (&TIME_CHAN == chan) {
+			enum time_status time_status;
+
+			err = zbus_chan_read(&TIME_CHAN, &time_status, K_FOREVER);
+			if (err) {
+				LOG_ERR("zbus_chan_read, error: %d", err);
+				SEND_FATAL_ERROR();
+				return;
+			}
+
+			if (time_status == TIME_AVAILABLE) {
+				LOG_DBG("Time available, sampling can start");
+
+				time_available = true;
+			}
+		}
+
+		if (time_available && (&TRIGGER_CHAN == chan)) {
```

Comment:
Wouldn't this make the module ignore `TRIGGER_CHAN` until time is available? So the first trigger (or multiple) might get ignored.

Reply:
Correct, it's just a more explicit way of doing the same as the `sample()` function already does, which is returning early without sampling when `date_time_uptime_to_unix_time_ms()` fails to get time

Reply:
Since multiple modules depend on time. Couldn't we just make the trigger module wait for TIME_AVAILABLE before publishing a trigger?

Reply:
+1.

Reply:
I considered it, but decided againsts it for a few reasons:
* Separation of concerns: Triggers and time are not really connected to each other. I think it's better to let the app module provide these "system wide" services like polling the shadow and distributing configuration, notifying the system about time and so on.
(It can be argued that the trigger module has an inherent dependency on the cloud module already, though, but that's a separate discussion that we can have at some point)
* Triggers have multiple users and not all depend on having time. Polling the shadow is independent of time and so is acquiring location. The dependencies in battery and environmental modules on having time should not block other modules from normal operation.
* Acquiring time can be slow and fail. If time is not sent by the network (common situation), NTP fails for some reason and GNSS time is not received (will not happen without a trigger first, so one of the potential time sources is blocked until time is available), sampling will be blocked.

Note that I'm not a big fan myself of the `time_available` flags that I introduce in this PR, but that is a short-lived thing as they will be replaced by states soon as part of CIA-1238.